processing/src/main/java/org/apache/druid/query/UnnestDataSource.java

```diff
@@ -172,14 +191,18 @@ public boolean equals(Object o)
       return false;
     }
     UnnestDataSource that = (UnnestDataSource) o;
-    return virtualColumn.equals(that.virtualColumn)
-           && base.equals(that.base);
+    if (unnestFilter != null) {
+      return virtualColumn.equals(that.virtualColumn) && unnestFilter.equals(that.unnestFilter)
+             && base.equals(that.base);
+    } else {
+      return virtualColumn.equals(that.virtualColumn) && base.equals(that.base);
+    }
```

Comment:
This is an odd equals implementation.  If the current object's `unnestFilter` is null, then the `that` object can have any unnest filter it wants?

Generally speaking, I suggest letting IntelliJ generate your equals and hashCode.  (I'm assuming that perhaps you adjusted it yourself?)

Reply:
yup, fixed